src/components/commit-list/CommitList.js

```diff
@@ -37,31 +38,39 @@ const Commit = ({ githubCommitObject }) => {
   );
 };
 
-export const CommitList = ({ commits, branches }) => {
+export const CommitList = ({ commits, branches, user, repo }) => {
+  const history = useHistory();
+
   const onSelect = (event) => {
-    // TODO: selecting a branch should reload the page to that branch
     event.preventDefault();
+    // convert to base64 to retain backslash characters
+    const branch = window.btoa(event.target.value);
+    history.push(`/user/${user}/repo/${repo}/branch/${branch}`);
   };
 
-  const SelectDropdown = () => (
-    <Flex justifyContent="flex-end" mb="25px">
-      <Select
-        borderColor="#808080"
-        color="#808080"
-        fontSize="15px"
-        onChange={onSelect}
-        w="160px"
-      >
-        {branches.map((branchName, index) => {
-          return (
-            <option key={index} value={branchName}>
-              {branchName}
-            </option>
-          );
-        })}
-      </Select>
-    </Flex>
-  );
+  const SelectDropdown = () => {
+    // TODO: add functionality for the edge case where a repo starts with main branch instead of master
+    return (
+      <Flex justifyContent="flex-end" mb="12px">
+        <Select
+          borderColor="#808080"
+          color="#808080"
+          fontSize="15px"
+          onChange={onSelect}
+          w="160px"
```

Comment:
We might need a value prop here. the Select always shows master as selected currently